core/templates/pages/topic-editor-page/topic-editor-auth.guard.ts

```diff
@@ -0,0 +1,65 @@
+// Copyright 2024 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Guard for the topic editor page.
+ */
+
+import {Location} from '@angular/common';
+import {Injectable} from '@angular/core';
+import {
+  ActivatedRouteSnapshot,
+  CanActivate,
+  Router,
+  RouterStateSnapshot,
+} from '@angular/router';
+
+import {AppConstants} from 'app.constants';
+import {AccessValidationBackendApiService} from 'pages/oppia-root/routing/access-validation-backend-api.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class TopicEditorAuthGuard implements CanActivate {
+  constructor(
+    private accessValidationBackendApiService: AccessValidationBackendApiService,
+    private router: Router,
+    private location: Location
+  ) {}
+
+  async canActivate(
+    route: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot
+  ): Promise<boolean> {
+    return new Promise<boolean>(resolve => {
+      let topicId = route.paramMap.get('topic_id') || '';
```

Comment:
What does the url look like when there is not topic_id param to get from the url? What happens when you send `''` as topic id for validation?

Reply:
Also, const instead of let

Reply:
> What does the url look like when there is not topic_id param to get from the url? What happens when you send `''` as topic id for validation?

It shows 404 page.
![image](https://github.com/user-attachments/assets/df0942c8-b741-48fc-96d0-0e0eb0456a54)

Done!!

Reply:
In that case, when route.paramMap.get('topic_id') give you an undefined or null then there is no point hitting the backend with an empty `''`. We can just show the 404 page from there itself. It will be faster and one less call to the servers.

Reply:
Done!